### PR TITLE
safeNextPathFrom(): update test cases

### PR DESCRIPTION
**Pull request by alxndrsn**

### CODE REVIEW

**Comment by matthew-white** [comment]
@alxndrsn, now that #1014 is merged, how are you thinking of this PR? I see new cases here that seem like they improve coverage (e.g., /login), as well as others that don't seem needed now that #1014 is merged (e.g., `/login/foo&sol;&period;&period;`).

**Comment by alxndrsn** [comment]
@matthew-white the `/login`-related tests actually indicate a proposed change of behaviour: should `safeNextPathFrom()` redirect `next=/login` to the web root?

**Comment by matthew-white** [comment]
> the `/login`-related tests actually indicate a proposed change of behaviour

I don't quite see the change. The test has:

```js
[ '/login',                          '/#/' ],
```

But isn't that what Backend already does? Or am I misunderstanding?

https://github.com/getodk/central-backend/blob/bab9ef971998fc29264cedbb02fe9c94de180920/lib/util/html.js#L27-L28

> should `safeNextPathFrom()` redirect `next=/login` to the web root?

No strong feelings! Unlike Frontend, Backend doesn't need special logic around `next=/login`. There's no problem if Backend redirects to /#/login. However, I can also see the case for keeping the logic around `next` the same in Frontend and Backend. Frontend redirects to /#/ for `next=/login`, and we could have Backend do the same. I think we should implement and test whichever way you think is best!

File: test/unit/util/html.js [hunk 1]
```diff
@@ -32,6 +32,12 @@ describe('util/html', () => {
       [ '/users',                 '/#/users' ],                   // eslint-disable-line no-multi-spaces
       [ '/users"><badTag ',       '/#/users%22%3E%3CbadTag' ],    // eslint-disable-line no-multi-spaces
 
+      // login URLs - in contrast to frontend
+      [ '/login',                          '/#/' ],    // eslint-disable-line no-multi-spaces
+      [ '/login/',                         '/#/' ],    // eslint-disable-line no-multi-spaces
+      [ '/login/foo/..',                   '/#/' ],    // eslint-disable-line no-multi-spaces
+      [ '/login/foo&sol;&period;&period;', '/#/' ],    // eslint-disable-line no-multi-spaces
```
[the remark this answers is not in the document]
Comment (alxndrsn) [reply test/unit/util/html.js]: I think this is now handled by `mustache`.  `&amp;` would be double-encoded as `&amp;amp;`.

**Comment by alxndrsn** [comment]
@matthew-white I've updated the PR to remove irrelevant tests, and add one for path traversal.

**Review by matthew-white — APPROVED** [review]

**Pull request merged**